docs/en/04_Changelogs/4.0.0.md

```diff
@@ -92,6 +93,138 @@ For a full list of renamed classes, check the `.upgrade.yml` definitions in each
 
 The rename won't affect class-based permission codes or database table names.
 
+##### Use ::class whenever possible
+
+Prefer:
+ ```php
+ private static $has_one = [
+   'MyObject' => MyObject::class
+ ];
+ ```
+ to
+ 
+ ```php
+private static $has_one = [
+  'MyObject' => 'My\Project\MyObject'
+];
+```
+
+##### Use Injector wherever you can
+
+This can help with reducing changes from 3.x to 4.x by allowing your code to 
+continue to use `MyClassName`, while you tell the injector what the FQ class name is:
+
+```yaml
+SilverStripe\Core\Injector\Injector:
+  MyClassName:
+    class: Me\MyModule\MyClassName # fully qualified!
+```
+
+##### Namespaced table names
+
+Keep in mind that table names are namespaced too. To ease data migrations, use the `$table_name` config property of your `DataObject` subclasses.
+This allows you to customise the table name to something simpler, or perhaps identical to its 3.x name.
+```php
+namespace SilverStripe\BannerManager;
+
+use SilverStripe\ORM\DataObject;
+
+class BannerImage extends DataObject
+{
+	private static $table_name = 'BannerImage';
+}
+```
+
+In order to ensure you are using the correct table for any class a new [DataObjectSchema](api:SilverStripe\ORM\DataObjectSchema) service
+is available to manage these mappings (see [versioned documentation](/developer_guides/model/data_model_and_orm)).
+
+
+```php
+public function countDuplicates($model, $fieldToCheck)
+{
+	$table = SilverStripe\ORM\DataObject::getSchema()->tableForField($model, $field);
+	$query = new SilverStripe\ORM\Queries\SQLSelect();
+	$query->setFrom("\"{$table}\"");
+	$query->setWhere(["\"{$table}\".\"{$field}\"" => $model->$fieldToCheck]);
+	return $query->count();
+}
+```
+
+##### Class name remapping
+
+If you've namespaced one of your custom page types, you may notice a message in the CMS 
+ telling you it's obsolete. This is likely because the `ClassName` 
+field in the `SiteTree` table still contains the singular model name, e.g. `BlogPost`
+and that when you change it to `SilverStripe\Blog\Model\BlogPost` then everything 
+works again.
+
+Luckily the `dev/build` task is configured to look for a legacy class name mapping 
```

Comment:
Not sure this is the best writing style for a technical upgrade manual. Nothing is "luckily" there, is it?

We also need to clarify the difference between class names changed in code (requires upgrader) and class names in database (this documentation block). The title "Class name remapping" doesn't explain that this is specific to database DBClassname values.

You may also need to clarify the distinction between user tasks and the explanation of the already implemented behaviour; This seems like the user is expected to write an upgrade yml, when in reality all they need to do is "run dev/build". The point of this block should focus on the "what users should do" rather than explaining the user story we took te implementing this feature.